mwdb/resources/object.py

```diff
@@ -76,14 +76,17 @@ def create_object(self, params):
 
         item, is_new = self._create_object(params, parent_object, share_with, metakeys)
 
-        db.session.commit()
+        try:
+            db.session.commit()
 
-        if is_new:
-            hooks.on_created_object(item)
-            self.on_created(item)
-        else:
-            hooks.on_reuploaded_object(item)
-            self.on_reuploaded(item)
+            if is_new:
+                hooks.on_created_object(item)
+                self.on_created(item)
+            else:
+                hooks.on_reuploaded_object(item)
+                self.on_reuploaded(item)
+        finally:
+            item.release_after_upload()
```

Comment:
Not a fan of `item.release_after_upload()`. Would it be possible to use context and `with` statement?

Reply:
I'm not a fan as well but that's why I'm using release method, we can't properly implement that using context manager.

Reply:
:cry: ok then